src/codex/CovarianceBlock.hpp

```diff
@@ -13,17 +13,24 @@ namespace codex {
 
   /* type aliases */
   template < typename T > using Matrix = Eigen::Matrix< T, Eigen::Dynamic, Eigen::Dynamic >;
+  template < typename T > using DiagonalMatrix = Eigen::DiagonalMatrix< T, Eigen::Dynamic >;
 
   /**
    *  @class
    *  @brief A covariance matrix block
    */
   class CovarianceBlock {
 
-    /* fields */
+    /* fields - meta data */
     MetaData row_;
     std::optional< MetaData > column_;
-    Matrix< double > matrix_;
+
+    /* fields - covariance matrix */
+    std::optional< Matrix< double > > covariances_;
+
+    /* fields - uncertainties and correlations */
+    std::optional< std::vector< double > > uncertainties_;
```

Comment:
Maybe replace 'uncertainties' by 'variances' vs 'standard_deviations'

Reply:
I picked standard deviations as the term to go with. If we want the variances specifically (i.e. the square of the standard deviations), we can just add a function to do so.